pkg/git.go

```diff
@@ -36,16 +35,39 @@ func NewGitPackage(source *spec.GitSource) Interface {
 }
 
 func (p *GitPackage) Install(ctx context.Context, dir, version string) (lockVersion string, err error) {
-	cmd := exec.CommandContext(ctx, "git", "clone", p.Source.Remote, dir)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err = cmd.Run()
-	if err != nil {
-		return "", err
+	_, err = os.Stat(dir)
+	if err != nil && os.IsNotExist(err) {
+		cmd := exec.CommandContext(ctx, "git", "clone", p.Source.Remote, dir)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
+		if err != nil {
+			return "", err
+		}
+	} else {
+		cmd := exec.CommandContext(ctx, "git", "remote", "set-url", "origin", p.Source.Remote)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		cmd.Dir = dir
+		err = cmd.Run()
+		if err != nil {
+			return "", err
+		}
+
+		cmd = exec.CommandContext(ctx, "git", "fetch", "origin")
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		cmd.Dir = dir
+		err = cmd.Run()
+		if err != nil {
+			return "", err
+		}
 	}
 
-	cmd = exec.CommandContext(ctx, "git", "checkout", version)
+	cmd := exec.CommandContext(ctx, "git", "checkout", "origin/"+version)
```

Comment:
Version is intentionally passed through directly so that any version, even "." works (if using the current directory/repo in CI for example). Generally the mean to do this is debatable on my side, just would like the use case to be possible (we could argue that such use cases are better suited with a "local" or "directory" source rather than a git source).

As I already use this in our CI pipeline I would prefer to first implement the alternative before merging this PR, if that's what we decide to do for my use case.

Reply:
> prefer to first implement the alternative before merging this PR

What alternative?  Happy to implement it...

Am I correct in understanding right now the jsonnetfile specifies either ".", "<sha>" or "<branch>" as version?  `git checkout <branch>` has the undesirable effect of not updating the local copy if you're already on that branch, hence the `origin/` prefix.